components/pages/explorer/RepoTable.tsx

```diff
@@ -229,19 +235,84 @@ const getTableStyle = (theme: typeof defaultTheme) => css`
   }
 `;
 
+const getDownloadError = (status: number, errText?: string) => {
+  switch (status) {
+    case 400:
+      if (errText === 'No files found.') {
```

Comment:
If you want to introduce error types, I wouldn't rely on message strings from the backend... these are liable to get changed because they just look like text to present to the user and not something used in application logic.

Instead, you may want to introduce some kind of ERROR_CODE enums/constants (documented on the API) for each unique kind of error you encounter and add that as an optional property in the error response. Sometimes we just display the error message from the backend instead.

Reply:
Yeah I had it set to display the backend errors initially, but talked to brandon and we decided to make the errors displayed more generic since the user isn't going to need specific details on the exact error, just that the download failed. 
But error codes sound like a better idea to handle this!